feature/experimental/bgp/ate_tests/base_bgp_session_parameters/base_bgp_session_parameters_test.go

```diff
@@ -0,0 +1,358 @@
+
+	capabilities := map[telemetry.E_BgpTypes_BGP_CAPABILITY]bool{
+		telemetry.BgpTypes_BGP_CAPABILITY_ROUTE_REFRESH: false,
+		telemetry.BgpTypes_BGP_CAPABILITY_ASN32:         false,
+		telemetry.BgpTypes_BGP_CAPABILITY_MPBGP:         false,
+	}
+	for _, cap := range nbrPath.SupportedCapabilities().Get(t) {
+		capabilities[cap] = true
+	}
+	for cap, present := range capabilities {
+		if !present {
+			t.Errorf("Capability not reported: %v", cap)
+		}
+	}
+
+}
+
+// verifyBgpTelemetry checks that the dut has an established BGP session with reasonable settings.
+func verifyBgpTelemetry(t *testing.T, dut *ondatra.DUTDevice) {
+	ifName := dut.Port(t, "port1").Name()
+	lastFlapTime := dut.Telemetry().Interface(ifName).LastChange().Get(t)
+	t.Log("Verifying BGP state")
+	statePath := dut.Telemetry().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(ateAttrs.IPv4)
+	nbr := statePath.Get(t).GetNeighbor(ateAttrs.IPv4)
+
+	// Get BGP adjacency state
+	t.Log("Waiting for BGP neighbor to establish...")
+	_, ok := nbrPath.SessionState().Watch(t, time.Minute, func(val *telemetry.QualifiedE_Bgp_Neighbor_SessionState) bool {
+		return val.IsPresent() && val.Val(t) == telemetry.Bgp_Neighbor_SessionState_ESTABLISHED
+	}).Await(t)
+	if !ok {
+		fptest.LogYgot(t, "BGP reported state", nbrPath, nbrPath.Get(t))
+		t.Fatal("No BGP neighbor formed")
+	}
+	status := nbrPath.SessionState().Get(t)
+	t.Logf("BGP adjacency for %s: %s", ateAttrs.IPv4, status)
+	if want := telemetry.Bgp_Neighbor_SessionState_ESTABLISHED; status != want {
+		t.Errorf("BGP peer %s status got %d, want %d", ateAttrs.IPv4, status, want)
+	}
+
+	// Check last established timestamp
+	lestTime := nbrPath.Get(t).GetLastEstablished()
+	t.Logf("BGP last est time :%v, flapTime :%v", lestTime, lastFlapTime)
+	if lestTime < lastFlapTime {
+		t.Errorf("Bad last-established timestamp: got %v, want >= %v", lestTime, lastFlapTime)
+	}
+
+	// Check BGP Transitions
+	estTrans := nbr.GetEstablishedTransitions()
+	t.Logf("Got established transitions: %d", estTrans)
+	if estTrans != 1 {
+		t.Errorf("Wrong established-transitions: got %v, want 1", estTrans)
+	}
+
+	// Check BGP neighbor address from telemetry
+	addrv4 := nbrPath.Get(t).GetNeighborAddress()
+	t.Logf("Got ipv4 neighbor address: %s", addrv4)
+	if addrv4 != ateAttrs.IPv4 {
+		t.Errorf("BGP v4 neighbor address: got %v, want %v", addrv4, ateAttrs.IPv4)
+	}
+	// Check BGP neighbor address from telemetry
+	peerAS := nbrPath.Get(t).GetPeerAs()
+	if peerAS != ateAS {
+		t.Errorf("BGP peerAs: got %v, want %v", peerAS, ateAS)
+	}
+
+	// Check BGP neighbor is enabled
+	if !nbrPath.Get(t).GetEnabled() {
+		t.Errorf("Expected neighbor %v to be enabled", ateAttrs.IPv4)
+	}
+}
+
+// Function to configure ATE configs based on args and returns ate topology handle.
+func configureATE(t *testing.T, ateParams *bgpTestParams) *ondatra.ATETopology {
```

Comment:
Sorry just realized this, since connType is only used by configureATE, move it from bgpTestParams to an argument of configureATE?

Also, rather than a string, let's use const like this:

```
type connType string

const (
  connInternal connType = "INTERNAL"
  connExternal connType = "EXTERNAL"
)
```

Reply:
updated, please check